pkg/koordlet/metricsadvisor/collector.go

```diff
@@ -123,7 +136,15 @@ func (c *collector) Run(stopCh <-chan struct{}) error {
 		return err
 	}
 
+	if c.haveGPU {
+		defer shutDownNvml()
+	}
+
 	go wait.Until(func() {
+		if c.haveGPU {
+			// collect gpu metrics.
+			c.collectProcessesGPUUsage()
```

Comment:
this check can be handle inside `collectProcessesGPUUsage`
return directly if does not have gpu